core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java

```diff
@@ -282,4 +299,27 @@ public void testThereCanBeOnlyOne() throws InterruptedException, ExecutionExcept
     }
   }
 
+  @Test
+  public void testHadoopCodecOverride() {
+    Algorithm.conf.set(Compression.Algorithm.CONF_LZ4_CLASS, DummyCodec.class.getName());
+    CompressionCodec dummyCodec = Compression.Algorithm.LZ4.createNewCodec(4096);
+    assertEquals("Hadoop override DummyCodec not loaded", DummyCodec.class, dummyCodec.getClass());
```

Comment:
These tests seem a bit heavyweight for what they are effectively testing. They are effectively testing that we can read the class name from the config files and/or system property. We don't actually need to create the codec to test that. These tests can probably be made more lightweight. However, given the triviality of my proposed one-liner above to get the class name, I'm not sure there's much to test here at all.

Reply:
Given the potential problems with JVM reuse and setting system properties, I'd probably just remove the test in this PR, but I agree with you in the general case about more unit testing for greater code coverage.
Yeah, I just don't want us to get to the point in our testing eagerness that we're doing things like: `assertTrue(Set.of(x).contains(x));`
At a certain point, we're not even testing Accumulo code, but that `conf.get()` and `System.getProperty()` work correctly.